planex/makesrpm.py

```diff
@@ -0,0 +1,160 @@
+#!/usr/bin/python
+
+"""
+planex-make-srpm: Wrapper around rpmbuild
+"""
+
+import sys
+import subprocess
+import os
+from shutil import copyfile, rmtree
+import fileinput
+import tarfile
+import tempfile
+
+import argparse
+import argcomplete
+from planex.util import add_common_parser_options
+
+
+def parse_args_or_exit(argv):
+    """
+    Parse command line options
+    """
+    parser = argparse.ArgumentParser(description='Planex SRPM building')
+    add_common_parser_options(parser)
+    parser.add_argument(
+        "--topdir", metavar="DIR", default=None,
+        help='Set rpmbuild toplevel directory')
+    parser.add_argument(
+        "--dist", metavar="DIST", default="",
+        help="distribution tag (used in RPM filenames)")
+    argcomplete.autocomplete(parser)
+    return parser.parse_known_args(argv)
+
+
+def parse_patchseries(series_file):
+    """
+    Parse series file and return the list of patches
+    """
+    patches = []
+    with open(series_file) as series:
+        for line in series:
+            line = line.partition('#')[0].strip()
+            if line:
+                patches.append(line)
+
+    return patches
+
+
+def setup_tmp_area():
+    """
+    Create temporary working area
+    """
+    tmp_dirpath = tempfile.mkdtemp()
+    tmp_specs = os.path.join(tmp_dirpath, 'SPECS')
+    tmp_build = os.path.join(tmp_dirpath, '_build')
+    tmp_sources = os.path.join(tmp_build, 'SOURCES')
+
+    os.makedirs(tmp_specs)
+    os.makedirs(tmp_build)
+    os.makedirs(tmp_sources)
+
+    return (tmp_dirpath, tmp_sources)
+
+
+def extract_patches(tmp_specfile, patchqueue_filters, patchqueue_path,
+                    tmp_sources, target):
+    """
+    Extract patches and inject them into the specfile
+    """
+    for line in fileinput.input(tmp_specfile, inplace=True):
+        if any([ext in line for ext in patchqueue_filters]):
+            tar = tarfile.open(patchqueue_path)
+            for mem in tar.getmembers():
+                mem.name = '%s-' % target + os.path.basename(mem.name)
```

Comment:
Could you please add a comment above this line to say that you are modifying mem.name in place to change the name of the extracted file (and drop any leading paths)?